akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -808,7 +808,7 @@ private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Mo
     }
   }
 
-  final def materialize(): Any = {
+  def materialize(): Any = {
```

Comment:
I think regarding the changes here, it might be better to see what is exactly the use case to see if we can factor things differently here instead of making non-final members. When dealing with inheritance I usually prefer methods/field to be either
 - private
 - public final
 - protected abstract

otherwise it becomes harder to reason about the state of the class.